### PR TITLE
React native upgrade

### LINKED ISSUE #136: Update react-native version
We are pretty far behind on react native versions. We should try to get as updated as possible

### PR DESCRIPTION
**Pull request by noahmalmed**
Fixes #136  .

Describe your changes.
- Updating react native version
- Updating all packages that depend on react native/ react version

Just a note on this PR: 
I would suggest going this commit by commit. A lot of the code changes are config changes done automatically by the upgrade script. A large portion of the code changes are from `prop-types` being moved into it's own package and out of reacts. I did my best to filter these big, "noisy" changes into their own commits.

# Review Checklist

- [x] Does it work in Android and iOS?
- [x] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] Are tests passing?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -25,25 +25,41 @@ main.jsbundle.meta
...
 local.properties
 zooniverse-key.keystore
 gradle.properties
+*.iml
 
 # node.js
 #
 node_modules/
 npm-debug.log
+yarn-error.log
```
Comment (wgranger) [inline .gitignore]: Do we need this? I don't believe we are using Yarn.

File: __mocks__/mockUtils.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+import PropTypes from 'prop-types';
+
+// Create a mock component
+export const createMockComponent = function (name) {
+    return class extends React.Component {
+      static displayName =name
```
Comment (wgranger) [inline __mocks__/mockUtils.js]: Pedantic, but can you put a space in here?

File: __mocks__/react-native-drawer.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import { createMockComponent } from './mockUtils'
+
+export const Drawer = createMockComponent('Drawer');
+export default Drawer;
```
Comment (wgranger) [inline __mocks__/react-native-drawer.js]: It seems odd having two exports here. I think you can remove the first.

File: __mocks__/react-native-image-pan-zoom.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import { createMockComponent } from './mockUtils'
+
+export const ImageZoom = createMockComponent('ImageZoom');
+export default ImageZoom;
```
Comment (wgranger) [inline __mocks__/react-native-image-pan-zoom.js]: Same as above.

**Comment by wgranger** [comment]
Take a look at a project with shortcuts (Floating Forests) on production through iOS and Android. The placement may be off in Android, and it doesn't appear that the text is appearing for either.

File: index.js [hunk 5]
```diff
@@ -10,19 +11,30 @@ import theme from './src/theme'
...
   render() {
+    
     return (
       <View style={styles.container}>
-        <StatusBar
-          barStyle="light-content"
-          hidden={false}
-          translucent={false}
-        />
+        <PlatformSpecificStatusBar />
```
Comment (wgranger) [inline index.js]: Is there a reason this component only renders for iOS?

Comment (noahmalmed) [reply index.js]: Not sure, it was basically the only difference between the two index files before I consolidated them.

Comment (wgranger) [reply index.js]: Ah, gotcha. Didn't notice this used to be the iOS index.

**Comment by wgranger** [comment]
Ok. Things look nice and updated (especially looking at the package.json. However, there is a blocking comment above with the shortcuts. Also, I noted on a few items I'm curious about. It was good to have this spin up without issues first try.

File: .gitignore (hunk 1 above)
Comment (noahmalmed) [reply .gitignore]: yeah, we can remove this. I think the update script added it.

**Comment by noahmalmed** [comment]
@wgranger changes made per your request. Everything should look good now. I actually also noticed that we weren't setting the color correctly for the toggle as well as not showing the text. Both of those should be fixed now.

**Comment by wgranger** [comment]
Cool. This LGTM.

**Pull request merged**